src/components/EditTextInPlace/index.tsx

```diff
@@ -0,0 +1,121 @@
+import Alert from '@material-ui/lab/Alert';
+import { ChangeEvent, useEffect, useRef, useState } from 'react';
+import { ClickAwayListener, FormControl, InputBase, InputLabel, Snackbar, Tooltip } from '@material-ui/core';
+import { FormattedMessage, useIntl } from 'react-intl';
+import { lighten, makeStyles } from '@material-ui/core/styles';
+
+export interface ComponentProps {
+    defaultText: string;
+    label: string;
+    onSubmit: (arg: string) => Promise<boolean>;
+    text: string;
+}
+
+const useStyles = makeStyles((theme) => ({
+    input: {
+        '&:focus, &:hover': {
+            borderColor: lighten(theme.palette.primary.main, 0.65 ),
+            paddingLeft: 10,
+            paddingRight: 0,
+        },
+        border: '2px dotted transparent',
+        borderRadius: 10,
+        paddingRight: 10,
+        transition: 'all 0.2s ease',
+    },
+    inputRoot: {
+        cursor: 'pointer',
+        fontFamily: 'inherit',
+        fontSize: 'inherit !important',
+        fontWeight: 'inherit',
+    },
+}));
+
+
+const EditTextinPlace: React.FunctionComponent<ComponentProps> = ({ label, text, onSubmit, defaultText }) => {
+    const [editing, setEditing] = useState<boolean>(false);
+    const [disabled, setDisabled] = useState<boolean>(false);
+    const [snackbar, setSnackbar] = useState<'success' | 'error'>();
+    const [newText, setNewText] = useState<string>(text);
+    const classes = useStyles();
+    const inputRef = useRef<HTMLInputElement>(null);
+    const intl = useIntl();
+
+    const intlIds = {
+        alert: `misc.components.editTextInPlace.alert.${snackbar || 'error'}`,
+        noEmpty: 'misc.components.editTextInPlace.noEmpty',
+        tooltip: `misc.components.editTextInPlace.tooltip.${editing ? 'save' : 'edit'}`,
+    };
+
+    useEffect(() => {
+        if (text !== newText) setNewText(text);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [text]);
+
+    const onRequestEdit = async () => {
+        await setEditing(true);
+        inputRef?.current?.focus();
+        if (text === defaultText) setNewText('');
+    };
+    const onCancelEdit = () => {
+        setEditing(false);
+        inputRef?.current?.blur();
+        setNewText(text);
+    };
+    const onChange = (evt: ChangeEvent<HTMLInputElement> ) => {
+        setNewText(evt.target.value);
+    };
+    const onKeyDown = (evt: React.KeyboardEvent) => {
+        if (evt.key === 'Enter' && !!newText) {
+            if ( newText === text) onCancelEdit();
+            else if ( newText !== text) submitChange();
+        }
+    };
+    const submitChange = (override?: string) => {
+        inputRef?.current?.blur();
+        setEditing(false);
+        setDisabled(true);
+        onSubmit(override || newText).then((success) => {
+            setSnackbar(success ? 'success' : 'error');
+            setDisabled(false);
+        });
```

Comment:
Sorry I realise that here is where the boolean state of the onSubmit it being used. I don't think it's a great pattern to use booleans for handling these error states so I would think about a different way to handle this... I don't know if the child handling the error of a function passed in from the parent is a good method in any case. Maybe @richardolsson has another opinion and thinks this is a good idea so interested what he thinks.

Reply:
Oh I just commented on that other thread about this. But yeah, I had that same concern when me and Everin first talked about it, and again when we talked about it all three of us, but then I dropped it because I realized this would be refactored with the stuff you're working on in #514 anyway.

I would be very happy to see this change, so that the error is handled outside (`onError` in the mutation, as you suggested).

Reply:
Even in my refector I think this wouldn't be fixed, because wherever the mutation would be implemented would require a return of `boolean` so I think it's worth figuring out a better way hmmmmm.

I wonder if the best way is to have an `error` prop which can be set from the parent. Then one could do `error={patchViewMutation.isError}`

Reply:
My thought about the refactor is that errors will probably be handled inside the code generated by the `createMutation()` factory, so we could then take out the error handling code entirely from this specific implementation.

But yeah, I agree that it should be refactored here, now. And I think the way to do it is to trigger the snackbar from `SingleViewLayout` rather than inside `EditTextInPlace`. Once we decide how to communicate errors to the user, I think that will be easier to implement.

If by then we want snackbars to be triggered consistently in response to errors, that can be handled in the factory. If we want snackbars only in this case, then doing it in `SingleViewLayout` is the right way anyway. And if we don't want snackbars at all, we can remove them here without risking breaking the `EditTextInPlace` which may then be used elsewhere.